src/index.js

```diff
@@ -9,188 +9,213 @@ moment.locale('fr')
+      })
 
-  // Get cookies from login page.
-  log('info', 'Get login form')
-  return (
-    request({
-      uri: 'https://login.orange.fr/',
-      resolveWithFullResponse
-    })
-      // Log in orange.fr
-      // eslint-disable-next-line no-unused-vars
-      .then(response => {
-        const { login, password } = fields
-
-        log('info', 'Send Orange login')
-        return request({
-          method: 'POST',
-          url: 'https://login.orange.fr/front/login',
-          body: {
-            login
-          },
-          resolveWithFullResponse
-          // eslint-disable-next-line no-unused-vars
-        }).then(response => {
-          log('info', 'Send Orange password')
-          return request({
-            method: 'POST',
-            url: 'https://login.orange.fr/front/password',
-            body: {
-              login,
-              password
-            }
-          })
-        })
+      const headers = {
+        'x-auth-id': response.headers['x-auth-id'],
+        'x-xsrf-token': response.headers['x-xsrf-token']
+      }
+      const { login, password } = fields
+
+      log('info', 'Send login with first XSRF token...')
+
+      response = await this.request({
+        method: 'POST',
+        url: 'https://login.orange.fr/front/login',
+        headers,
+        body: {
+          login
+        },
+        resolveWithFullResponse
       })
-      .then(body => {
-        if (body.credential != null || body.password != null) {
-          throw new Error(body.credential || body.password)
+
+      headers['x-xsrf-token'] = response.headers['x-xsrf-token']
+      log('info', 'Send password with second XSRF token...')
+
+      const body = await this.request({
+        method: 'POST',
+        url: 'https://login.orange.fr/front/password',
+        headers,
+        body: {
+          login,
+          password
         }
       })
-      .catch(err => {
-        log('error', err)
-        if (err && err.message.includes('bloqué')) {
-          throw new Error('LOGIN_FAILED.TOO_MANY_ATTEMPTS')
-        } else if (err && err.message.includes('un problème technique')) {
-          throw new Error(errors.VENDOR_DOWN)
-        } else if (err.statusCode === 401) {
-          throw new Error(errors.LOGIN_FAILED)
-        } else {
-          throw new Error(errors.VENDOR_DOWN)
+
+      if (body.credential != null || body.password != null) {
+        throw new Error(body.credential || body.password)
+      }
+    } catch (err) {
+      log('error', err)
+      if (err && err.message.includes('bloqué')) {
+        throw new Error('LOGIN_FAILED.TOO_MANY_ATTEMPTS')
+      } else if (err && err.message.includes('un problème technique')) {
+        throw new Error(errors.VENDOR_DOWN)
+      } else if (err.statusCode === 401) {
+        throw new Error(errors.LOGIN_FAILED)
+      } else {
+        throw new Error(errors.VENDOR_DOWN)
+      }
+    }
+    log('info', 'Successfully logged in.')
+  }
+
+  async fetchPage() {
```

Comment:
async to remove

Reply:
Async was remove from parsePage. this function need it as it contains request
Was just a swap mistake between fetchPage and parsePage